src/rp2040/hardware_structs/include/hardware/structs/i2c.h

```diff
@@ -10,125 +12,320 @@
     io_rw_32 dma_tdlr;
+
+    _REG_(I2C_IC_DMA_RDLR_OFFSET)
+    // I2C Receive Data Level Register
+    // 0x0000000f [0-3]   : DMARDL (0): Receive Data Level
     io_rw_32 dma_rdlr;
+
+    _REG_(I2C_IC_SDA_SETUP_OFFSET)
+    // I2C SDA Setup Register
+    // 0x000000ff [0-7]   : SDA_SETUP (0x64): SDA Setup
     io_rw_32 sda_setup;
+
+    _REG_(I2C_IC_ACK_GENERAL_CALL_OFFSET)
+    // I2C ACK General Call Register
+    // 0x00000001 [0]     : ACK_GEN_CALL (1): ACK General Call
     io_rw_32 ack_general_call;
-    io_rw_32 enable_status;
+
+    _REG_(I2C_IC_ENABLE_STATUS_OFFSET)
+    // I2C Enable Status Register
+    // 0x00000004 [2]     : SLV_RX_DATA_LOST (0): Slave Received Data Lost
+    // 0x00000002 [1]     : SLV_DISABLED_WHILE_BUSY (0): Slave Disabled While Busy (Transmit, Receive)
+    // 0x00000001 [0]     : IC_EN (0): ic_en Status
+    io_ro_32 enable_status;
+
+    _REG_(I2C_IC_FS_SPKLEN_OFFSET)
+    // I2C SS, FS or FM+ spike suppression limit
+    // 0x000000ff [0-7]   : IC_FS_SPKLEN (0x7): This register must be set before any I2C bus transaction can take place to...
     io_rw_32 fs_spklen;
+
     uint32_t _pad2;
-    io_rw_32 clr_restart_det;
+
+    _REG_(I2C_IC_CLR_RESTART_DET_OFFSET)
+    // Clear RESTART_DET Interrupt Register
+    // 0x00000001 [0]     : CLR_RESTART_DET (0): Read this register to clear the RESTART_DET interrupt (bit 12) of...
+    io_ro_32 clr_restart_det;
+
+    uint32_t _pad3[18];
+
+    _REG_(I2C_IC_COMP_PARAM_1_OFFSET)
+    // Component Parameter Register 1
+    // 0x00ff0000 [16-23] : TX_BUFFER_DEPTH (0): TX Buffer Depth = 16
+    // 0x0000ff00 [8-15]  : RX_BUFFER_DEPTH (0): RX Buffer Depth = 16
+    // 0x00000080 [7]     : ADD_ENCODED_PARAMS (0): Encoded parameters not visible
+    // 0x00000040 [6]     : HAS_DMA (0): DMA handshaking signals are enabled
+    // 0x00000020 [5]     : INTR_IO (0): COMBINED Interrupt outputs
+    // 0x00000010 [4]     : HC_COUNT_VALUES (0): Programmable count values for each mode
+    // 0x0000000c [2-3]   : MAX_SPEED_MODE (0): MAX SPEED MODE = FAST MODE
+    // 0x00000003 [0-1]   : APB_DATA_WIDTH (0): APB data bus width is 32 bits
+    io_ro_32 comp_param_1;
+
+    _REG_(I2C_IC_COMP_VERSION_OFFSET)
+    // I2C Component Version Register
+    // 0xffffffff [0-31]  : IC_COMP_VERSION (0x3230312a)
+    io_ro_32 comp_version;
+
+    _REG_(I2C_IC_COMP_TYPE_OFFSET)
+    // I2C Component Type Register
+    // 0xffffffff [0-31]  : IC_COMP_TYPE (0x44570140): Designware Component Type number = 0x44_57_01_40
+    io_ro_32 comp_type;
 } i2c_hw_t;
 
 #define i2c0_hw ((i2c_hw_t *const)I2C0_BASE)
 #define i2c1_hw ((i2c_hw_t *const)I2C1_BASE)
 
-// List of configuration constants for the Synopsys I2C hardware (you may see
-// references to these in I2C register header; these are *fixed* values,
-// set at hardware design time):
-
-// IC_ULTRA_FAST_MODE ................ 0x0
-// IC_UFM_TBUF_CNT_DEFAULT ........... 0x8
-// IC_UFM_SCL_LOW_COUNT .............. 0x0008
```

Comment:
Eeek, see https://github.com/raspberrypi/pico-feedback/issues/191 - this  definitely needs to be moved "somewhere else".

Reply:
ugh - yes this should not have been here, perhaps it was _hard_ to put it in the regs header where it really belongs ... can we do that now (presumably upstream even of the regs header)

Reply:
The regs header is also autogenerated, but I'll have a think about how to include this comment-text.

Reply:
Why not have it in a platform def type file. Could be in the main platform_defs.h or in a specific, i2c_defs.h. Then we don't need to hack the autogenerated files and it can be included by the i2c driver

Reply:
Also now fixed in #601 :grin: